### PR TITLE
bench: Include constraints progession performance

### LINKED ISSUE #238: Add more descriptive benchmarks
The benchmarks should ideally :

- compare against other well known proving systems, so that others can understand what the relative performance is.

- Compare against different circuit sizes.

- Since the ratio of addition gates to multiplication gates, plays a part in the benchmarks, [See PLONK paper] we should also give the worse case and best case ratio.

There are also use-case specific benchmarks which may be useful, such as Pedersen hashes and SHA256 circuits in each proving system.

This is a bit more in depth and does not hold high priority at the moment. However, in the end I think it would be good to give the end user a quick summary of whether this library could fit into their stack.

### PR DESCRIPTION
**Pull request by vlopes11**
The expected verification time of zk-Snarks is constant while the
proving time will increase depending on the number of used constraints.

This benchmark will show the performance progression without public
inputs.

Resolves #238

### CODE REVIEW

**Review by Bounce23 — CHANGES_REQUESTED** [review]
Overall it looks very good to me. One thing that may help is syntax change, perhaps its best to add in the numbers in terms of powers of two, as with the circuit padding, this format is most relatable.

**Review by Bounce23 — APPROVED** [review]
LGTM!

**Comment by CPerezz** [comment]
Let's wait and see if #511 provides decent-enough results. If that's the case, I'll rebase it to this branch and change the Head to be this branch too. Merge here and then all together to `release-0.8`.

Is it ok @ZER0 @vlopes11 ?

File: README.md [hunk 1]
```diff
@@ -130,11 +130,43 @@ There are two main types of documentation in this repository:
...
+2^9  = 512 constraints     time:   [15.047 ms 15.181 ms 15.471 ms]
+2^10 = 1024 constraints    time:   [15.630 ms 15.888 ms 16.303 ms]
+2^11 = 2048 constraints    time:   [15.524 ms 15.719 ms 15.913 ms]
+2^12 = 4096 constraints    time:   [15.178 ms 15.445 ms 15.593 ms]
+2^13 = 8192 constraints    time:   [15.437 ms 15.874 ms 16.553 ms]
+2^14 = 16384 constraints   time:   [16.059 ms 16.386 ms 16.608 ms]
+2^15 = 32768 constraints   time:   [15.969 ms 16.357 ms 16.660 ms]
+2^16 = 65536 constraints   time:   [16.233 ms 16.462 ms 16.639 ms]
+2^17 = 131072 constraints  time:   [18.213 ms 18.680 ms 18.886 ms]
+```
```
Comment (CPerezz) [inline README.md:169]: Could you run again the benches after the lates additions and add back the new results?

**Review by CPerezz — APPROVED** [review]
I would like to add the new results. Afther the performance regression fixes. For the rest, LGTM.

**Pull request merged**